core/src/main/scala/io/iteratee/internal/Step.scala

```diff
@@ -108,7 +108,7 @@ final object Step { self =>
         case Done(otherValue, otherRemaining) => F.pure(Done(otherValue, otherRemaining ++ remaining))
         case step =>
           val c = remaining.lengthCompare(1)
-          if (c < 0) F.pure(step) else if (c == 0) step.feedEl(remaining.head) else step.feedNonEmpty(remaining)
+          if (c < 0) f(value) else if (c == 0) step.feedEl(remaining.head) else step.feedNonEmpty(remaining)
```

Comment:
Let's put this back to the way it was before. Below, I will modify the test I put in with this change.